v2/controllers/pod_watcher.go

```diff
@@ -178,23 +193,61 @@ OUTER2:
 	}
 
 	r.podAddrs[key] = podIPs
+	for _, ip := range podIPs {
+		keySet, ok := r.peers[ip.String()]
+		if !ok {
+			keySet = map[string]struct{}{}
+			r.peers[ip.String()] = keySet
+		}
+		keySet[key] = struct{}{}
+	}
+
 	r.metric.Set(float64(len(r.podAddrs)))
 	return nil
 }
 
+func (r *podWatcher) delTerminatedPod(pod *corev1.Pod, logger logr.Logger) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	return r.delPeer(fmt.Sprintf("%s/%s", pod.Namespace, pod.Name), "delTerminatedPod", string(pod.Status.Phase), logger)
+}
+
 func (r *podWatcher) delPod(n types.NamespacedName, logger logr.Logger) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	key := n.Namespace + "/" + n.Name
+	return r.delPeer(fmt.Sprintf("%s/%s", n.Namespace, n.Name), "delPod", "", logger)
```

Comment:
```suggestion
        return r.delPeer(pod.Namespace + "/" + pod.Name, "delPod", "", logger)
```

I think it is good to me to use whether ways `fmt` package or `+` operator.
But I think it is better to unify.